app/controllers/KairosDbProxyController.java

```diff
@@ -165,6 +173,12 @@ public KairosDbProxyController(
                         Results.badRequest("All queried metrics must have at least one aggregator"));
             }
 
+            if (_addMergeAggregator) {
+                return _kairosService.queryMetrics(addMergeAggregator(metricsQuery))
```

Comment:
nit: Should we make `metricsQuery` non-final and change it here (e.g. `metrics=addMergeAggregator(metricsQuery)`) and then let the normal flow resume?